fhir-server/src/main/java/au/csiro/pathling/fhirpath/function/ReverseResolveFunction.java

```diff
@@ -48,27 +48,28 @@ public FhirPath invoke(@Nonnull final NamedFunctionInput input) {
 
     // Check that the argument types include the input type.
     final Set<ResourceType> argumentTypes = ((ReferencePath) argument).getResourceTypes();
-    final ResourceType inputType = ((ResourcePath) inputPath).getResourceType();
+    final ResourceType inputType = inputPath.getResourceType();
     checkUserInput(argumentTypes.contains(inputType),
         "Reference in argument to reverseResolve does not support input resource type: "
             + expression);
 
     // Do a left outer join from the input to the argument dataset using the reference field in the
```

Comment:
This most likely is OK but the comment is confusing ...

Reply:
See bbe14269aadd6b9290b98bcc353a3d8b4fb9cb4f.